nautobot/ipam/tests/test_models.py

```diff
@@ -399,6 +399,18 @@ def test_duplicate_nonunique_role(self):
             role=IPAddressRoleChoices.ROLE_VIP,
         )
 
+    def test_multiple_nat_outside(self):
+        nat_inside = IPAddress.objects.create(address=netaddr.IPNetwork("192.168.0.1/24"))
+        nat_outside1 = IPAddress.objects.create(address=netaddr.IPNetwork("192.0.2.1/24"), nat_inside=nat_inside)
+        nat_outside2 = IPAddress.objects.create(address=netaddr.IPNetwork("192.0.2.2/24"), nat_inside=nat_inside)
+
+        nat_inside.refresh_from_db()
+
+        with self.assertRaises(IPAddress.NatOutsideMultipleObjectsReturned):
+            nat_inside.nat_outside
+        self.assertEqual(nat_inside.nat_outside_list.first(), nat_outside1)
+        self.assertEqual(nat_inside.nat_outside_list.last(), nat_outside2)
+
```

Comment:
Also test that the `nat_outside` setter works as expected in the cases of:
- two existing nat_outside entries (exception)
- one existing nat_outside entry (replace it, I think?)

Reply:
👍 I knew this needed a bit of better test coverage but wanted to get the commit up last night to start a review of the solution.